proposed/2022/vulnerabilities-in-restore.md

```diff
@@ -0,0 +1,406 @@
+
+We ran a survey in 2022 regarding how developers currently feel about known security advisories and here is what we've learned:
+
+74% of developers would like to be notified of packages at restore/build time. 65% would like to see icons in the solution explorer for problematic packages. 42% of developers would like a security notification in Visual Studio's notification window. 39% of developers would use a dotnet audit command.
+
+![Notified](./../../meta/resources/VulnerabilitiesInRestore/Notified.png)
+
+Next, we asked developers how they learned about security vulnerabilities in their software. Majority learned from public reports, 3rd-party vulnerability scanners, and code reviews/accidental discovery.
+
+![Learned](./../../meta/resources/VulnerabilitiesInRestore/Learned.png)
+
+The most challenging aspects for ensuring the software they write is secure is three major things:
+
+1. Evaluating third-party libraries are secure and free of vulnerabilities.
+2. Verifying the software they write is free of vulnerabilities.
+3. Applying best practices to writing code.
+
+![Challenging](./../../meta/resources/VulnerabilitiesInRestore/Challenging.png)
+
+When asked how they understood and resolved the vulnerability, the majority of developers resolved them by updating the dependency or app.
+
+![Resolved](./../../meta/resources/VulnerabilitiesInRestore/Resolved.png)
+
+When doing so, the most challenging aspect for them is not breaking something else when doing this and better understanding the vulnerability's information to assess the impact and root cause to see if they are actually affected.
+
+![ResolveChallenge](./../../meta/resources/VulnerabilitiesInRestore/ResolveChallenge.png)
+
+## Explanation
+
+### Functional explanation
+
+When a restore happens whether implicitly, or explicitly, the packages will be audited for vulnerabilities *if* any of the sources declared provide vulnerability information.
+If any vulnerabilities are found, the impact, appropriate remediation, and other advisory metadata will be reported to the developer to take action on.
+
+For example, a developer might run `dotnet restore` explicitly or `dotnet add package` which runs restore implicitly. If a package being restored contains a known security advisory, they would see output at the bottom of the restore output that indicates how many vulnerabilities, a break-down of their severities, and the appropriate remediation action to take given the context of their environment. Below are a handful of concepts ranging from a low to high verbosity of information
+
+#### Warning codes
+
+| Warning Code | Severity |
+|--------------|----------|
+| NU1901 | low |
+| NU1902 | moderate |
+| NU1903 | high |
+| NU1904 | critical |
+
+When there are packages used by the project with vulnerabilities, they will be outjust like like any other warning or error:
+
+```text

```

Comment:
If you put bash, you get better highlighting :) Not a big deal :D

Reply:
The only highlighting I see are the words `in`, `for` and the `<` and `>` characters. Which make sense in bash scripts, but in my opinion it doesn't make sense for these samples.  Why in the sentence "Found 2 vulerabilities in 1 page(s)" should only the word `in`, be highlighted?  🤷

Reply:
Here's what I am seeing: 

![image](https://user-images.githubusercontent.com/2878341/218845920-fde7720d-77f8-4427-a761-62c7a83c82ff.png)

The quoted part is being highlighted. It's not a big deal either way, and most definitely a nit :) Your call.